dataflow/flex-templates/pipeline_with_dependencies/my_package/my_transforms.py

```diff
@@ -0,0 +1,37 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Defines custom PTransforms and DoFns used in the pipleines."""
+
+import re
+from typing import Iterable
+
+import apache_beam as beam
+
+
+class WordExtractingDoFn(beam.DoFn):
+    """Parses each line of input text into words."""
+
+    def process(self, element: str) -> Iterable[str]:
+        return re.findall(r'[\w\']+', element, re.UNICODE)
+
+
+class FindLongestWord(beam.PTransform):
```

Comment:
Nit: could this be simpler with the [`@ptransform_fn`](https://beam.apache.org/releases/pydoc/current/apache_beam.transforms.ptransform.html#apache_beam.transforms.ptransform.ptransform_fn) decorator?